presto-hudi/src/test/java/com/facebook/presto/hudi/AbstractHudiDistributedQueryTestBase.java

```diff
@@ -0,0 +1,380 @@
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Streams;
+import com.google.common.io.Resources;
+import org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat;
+import org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe;
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.hadoop.HoodieParquetInputFormat;
+import org.apache.hudi.hadoop.realtime.HoodieParquetRealtimeInputFormat;
+import org.testng.annotations.AfterClass;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static com.facebook.presto.hive.HiveType.HIVE_BINARY;
+import static com.facebook.presto.hive.HiveType.HIVE_BOOLEAN;
+import static com.facebook.presto.hive.HiveType.HIVE_DATE;
+import static com.facebook.presto.hive.HiveType.HIVE_DOUBLE;
+import static com.facebook.presto.hive.HiveType.HIVE_FLOAT;
+import static com.facebook.presto.hive.HiveType.HIVE_INT;
+import static com.facebook.presto.hive.HiveType.HIVE_LONG;
+import static com.facebook.presto.hive.HiveType.HIVE_STRING;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+public abstract class AbstractHudiDistributedQueryTestBase
+        extends AbstractTestQueryFramework
+{
+    public static final String HUDI_CATALOG = "hudi";
+
+    public static final String HUDI_SCHEMA = "testing"; // Schema in Hive which has test hudi tables
+
+    protected static ExtendedHiveMetastore metastore;
+
+    private static final String OWNER_PUBLIC = "public";
+    protected static final MetastoreContext METASTORE_CONTEXT = new MetastoreContext("test_user", "test_queryId", Optional.empty(), Optional.empty(), Optional.empty(), false, HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER_PROVIDER);
+    private static final PrincipalPrivileges PRINCIPAL_PRIVILEGES = new PrincipalPrivileges(ImmutableMultimap.of(), ImmutableMultimap.of());
+
+    private static final StorageFormat STORAGE_FORMAT_COPY_ON_WRITE = StorageFormat.create(
+            ParquetHiveSerDe.class.getName(),
+            HoodieParquetInputFormat.class.getName(),
+            MapredParquetOutputFormat.class.getName());
+    private static final StorageFormat STORAGE_FORMAT_MERGE_ON_READ = StorageFormat.create(
+            ParquetHiveSerDe.class.getName(),
+            HoodieParquetRealtimeInputFormat.class.getName(),
+            MapredParquetOutputFormat.class.getName());
+
+    //     spark.sql(
```

Comment:
Wondering if it's time to introduce the Java write client for testing purposes, instead of simulating commits this way. We already do it that way in Trino. I am ok with this change. We can take it up as a followup. But what do you think?

Reply:
yes, i will try 
thanks